app/controllers/boards_controller.rb

```diff
@@ -40,6 +40,7 @@ def show
     @action_item = ActionItem.new(board_id: @board.id)
     @board_creators = User.find(@board.memberships.where(role: 'creator').pluck(:user_id))
                           .pluck(:email)
+    @current_creator = current_user&.id if @board_creators.include?(current_user&.email)
```

Comment:
Why do we need @current_creator at all? I thought that if you have correct policy it should be enough to make it all work